[1.13] Instruct users to manually clean stale routing rules after downgrade

jschwinger233 opened the pull request:
Downgrade from 1.13 to 1.12 (EOL) requires manual cleanups, this PR adds the upgrade and downgrade notes in Documentation.

This PR also makes sure the manual cleanups are only run for minor downgrade in ci-ipsec-upgrade.

viktor-kurchenko reviewed (approved):
LGTM

julianwiedmann commented on Documentation/security/network/encryption-ipsec.rst:
```diff
@@ -193,6 +193,25 @@ to all clusters in the mesh. You might need to increase the transition time to
 allow for the new keys to be deployed and applied across all clusters,
 which you can do with the agent flag ``ipsec-key-rotation-duration``.
 
+Downgrade
+=========
+
+.. attention::
+
+   This is a v1.13-only requirement. In v1.14 and later, the Cilium agent can
+   automatically remove stale routing rules.
+
+To ensure connectivity via L7/DNS proxy after downgrade, please run commands
+after downgrade completes:
+
+.. code-block:: shell-session
+
+    cilium_pods=$(kubectl -nkube-system get po -l k8s-app=cilium --no-headers -o custom-columns=":metadata.name")
+    for cilium_pod in $cilium_pods; do
+        kubectl -nkube-system exec $cilium_pod -- ip -4 rule delete fwmark 0xB00/0xF00 lookup 2005 2>/dev/null || true
+        kubectl -nkube-system exec $cilium_pod -- ip -6 rule delete fwmark 0xB00/0xF00 lookup 2005 2>/dev/null || true
+    done
+
```
I wonder if we should also call this out in https://github.com/cilium/cilium/blob/v1.13/Documentation/operations/upgrade.rst (or *only* there?).

@qmonnet mind giving the change a look from doc perspective? :pray:

qmonnet replied:
It's a good idea to have it in upgrade.rst as well I'd say, and it doesn't hurt to have it at both locations. Not something we'll support for long anyway given that 1.13 goes away soon.

julianwiedmann replied:
looks good now, ty @jschwinger233 !

julianwiedmann reviewed (approved):
workflow change looks as expected, thank you! Just one comment on the doc change.

qmonnet reviewed (approved):
Looks good to me, thanks!

Please next time don't forget to provide context and motivation for the change in the commit description (in particular for the first commit).

jschwinger233 committed:
ci-ipsec-upgrade: Run manual cleanups only for minor downgrade

Recent security fixes require some new ip routing rules like "fwmark
0xb00/0xf00 lookup 2005" in 1.13, which leave a downgrade problem when
users want to rollback after an unsuccessful 1.12->1.13 upgrade. Because
1.12 is EOL, there is no way to do graceful downgrade automatically, we
should run some commands to delete those stale routing rules in old
1.12 cilium.

Those commands are also set in ci-ipsec-upgrade to ensure healthy CI
status. As we explained above, this manual step is necessary only
for 1.13->1.12 downgrade (minor downgrade). This patch therefore runs
manual cleanups only for minor downgrade.

Suggested-by: Julian Wiedmann <jwi@isovalent.com>
Signed-off-by: gray <gray.liang@isovalent.com>

jschwinger233 committed:
doc: Instruct users to manually clean stale routing after downgrade

Signed-off-by: gray <gray.liang@isovalent.com>
